minio/signer.py

```diff
@@ -74,6 +74,7 @@ def presign_v4(method, url, credentials,
     :param expires: final expiration of the generated URL. Maximum is 7days.
     :param response_headers: Specify additional query string parameters.
     :param request_date: the date of the request.
+    :param content_hash_hex: the sha256sum of the payload if already calcualted.
```

Comment:
can we call this `content_sha256_hex` instead to be more accurate?

Reply:
Sure - is that the only kind that it can be though? Note that if `X-Amz-Algorithm` is something different, or another hash header is provided to sign, why would the variable restrict to sha256? note that the variable already existed in the function and had the name `content_hash_hex` which is why I kept it like that.

Reply:
Yes in S3 that is always going to be sha256 unless entire AWS S3 changes.

Reply:
when that time comes we need to write a new SDK @vsoch

Reply:
haha okay, let me change that quickly then :)

Reply:
So to keep it generic we already take headers so if that covers it, then we don't need additional param - just honor `x-amz-content-sha256 ` from input headers.